### PR TITLE
Added support for vertx Future when client fails to connect.

### PR DESCRIPTION
**Pull request by w00lf**
#46

### CODE REVIEW

File: src/test/java/io/vertx/rabbitmq/RabbitMQServiceTest.java [hunk 1]
```diff
@@ -31,43 +34,40 @@
...
 
+  @Rule
+  public SniffyRule sniffyRule = new SniffyRule();
+
+  @Test
+  @DisableSockets
+  public void testStartWithReconnectFuture() throws Exception {
+    Future<Void> future = Future.future();
+    CountDownLatch latch = new CountDownLatch(1);
+    RabbitMQOptions options = "true".equalsIgnoreCase(System.getProperty("rabbitmq.local")) ? config().setUri(CLOUD_AMQP_URI) : config();
```
Comment (thibaultleouay) [inline src/test/java/io/vertx/rabbitmq/RabbitMQServiceTest.java]: Shouldnt it be : 

"true".equalsIgnoreCase(System.getProperty("rabbitmq.local")) ? config() : config().setUri(CLOUD_AMQP_URI)

**Comment by w00lf** [comment]
@thibaultleouay fixed your notice.

**Comment by vietj** [comment]
can you merge it and set milestone to 3.5.1-SNAPSHOT ?

**Comment by vietj** [comment]
actually contributed by myself here https://github.com/vert-x3/vertx-rabbitmq-client/issues/46

**Pull request closed without merging**